cmake/RuntimeChecks.cmake

```diff
@@ -106,9 +106,17 @@ elseif(RUNTIME_CHECK STREQUAL "asan")
     message(FATAL_ERROR "libubsan not installed - address sanitizer not available")
   endif(UBSAN_LIBRARY-NOTFOUND)
   message(STATUS "Runtime memory checker: gcc/clang address sanitizers")
+  option(SANITIZE_3RD_PARTY "Detect leaks in 3rd party libraries used by Dispatch while running tests" OFF)
+  file (COPY "${CMAKE_SOURCE_DIR}/tests/lsan.supp" DESTINATION "${CMAKE_BINARY_DIR}/tests")
```

Comment:
Instead of copying to the build folder can we keep the lsan.supp in the source folder but add it to the .gitignore ? Then add a file called lsan.supp.in file to the tests folder and add that to git.

Reply:
I saw that pattern used elsewhere in dispatch. I thought that having generated files outside of `CMAKE_BINARY_DIR` is discouraged.

One problem with that is when you have multiple `build` folders, e.g. debug and release build. Then running cmake for both at approx the same time (my IDE does that) can cause weird issues.

I guess I can do as you ask. Also, I should probably try to use `configure_file` in cmake.

Reply:
Good point about not having generated files outside of CMAKE_BINARY_DIR. I remember following that rule when working in Java projects using Maven. We don't seem to follow that rule here which I have not questioned so far. 
And, yes, I was talking about using the configure_file in cmake.
But you should wait on this to see what @kgiusti 's take on this is. Thx

Reply:
`configure_file` does not work all that well for my purposes; I want to essentially call a `cat` to concatenate suppression lists together, so I used a custom command